analytics_data_api/v0/views/__init__.py

```diff
@@ -97,3 +108,119 @@ def finalize_response(self, request, response, *args, **kwargs):
         if request.META.get('HTTP_ACCEPT') == u'text/csv':
             response['Content-Disposition'] = u'attachment; filename={}'.format(self.get_csv_filename())
         return super(CsvViewMixin, self).finalize_response(request, response, *args, **kwargs)
+
+
+class APIListView(generics.ListAPIView):
+    """
+    An abstract view to store common code for views that return a list of data.
+
+    **Example Requests**
+
+        GET /api/v0/some_endpoint/
+            Returns full list of serialized models with all default fields.
+
+        GET /api/v0/some_endpoint/?ids={id},{id}
+            Returns list of serialized models with IDs that match an ID in the given
+            `ids` query parameter with all default fields.
+
+        GET /api/v0/some_endpoint/?ids={id},{id}&fields={some_field},{some_field}
+            Returns list of serialized models with IDs that match an ID in the given
+            `ids` query parameter with only the fields in the given `fields` query parameter.
+
+        GET /api/v0/some_endpoint/?ids={id},{id}&exclude={some_field},{some_field}
+            Returns list of serialized models with IDs that match an ID in the given
+            `ids` query parameter with all fields except those in the given `exclude` query
+            parameter.
+
+    **Response Values**
+
+        Since this is an abstract class, this view just returns an empty list.
+
+    **Parameters**
+
+        This view supports filtering the results by a given list of IDs. It also supports
+        explicitly specifying the fields to include in each result with `fields` as well of
+        the fields to exclude with `exclude`.
+
+        ids -- The comma-separated list of identifiers for which results are filtered to.
+            For example, 'edX/DemoX/Demo_Course,course-v1:edX+DemoX+Demo_2016'. Default is to
+            return all courses.
+        fields -- The comma-separated fields to return in the response.
+            For example, 'course_id,created'. Default is to return all fields.
+        exclude -- The comma-separated fields to exclude in the response.
+            For example, 'course_id,created'. Default is to not exclude any fields.
+    """
+    ids = None
+    fields = None
+    exclude = None
+    always_exclude = []
+    serializer_class = serializers.ModelSerializer
+    model = models.Model
+    model_id = 'id'
```

Comment:
:+1: naming is hard, definitely keep letting me know if you have better variable name suggestions.